r/R/embedding.R

```diff
@@ -113,9 +113,9 @@ assignEmbedding <- function(embedding_data, data, reduction_method = "umap") {
   embedding <- embedding[cells_id + 1, ]
   rownames(embedding) <- colnames(data)
 
-  embedding_key = "UMAP_"
-  if(reduction_method == "tsne") {
-    embedding_key = "tSNE_"
+  embedding_key <- "UMAP_"
+  if (reduction_method == "tsne") {
+    embedding_key <- "tSNE_"
```

Comment:
I don't like this if a lot from a style point of view. You could use a dictionary  and index it with the reduction method. e.g.
```
const reductionKeys = {
    'umap': 'UMAP_",
    'tsne': 'tSNE_'
}

and then do `embedding_key = reduction_keys[reduction_method]`

or if the lowecase `t` in `tSNE` is not needed just turn the method into uppercase and add the underscore something like: 

`embedding_key <- uppercase(reduction_method) + '_'`

Reply:
Agree on the named list! The low case "t" is the usual way t-SNE is stylized. I don't have strong opinions, but would check what Seurat does by default.

Regarding the underscore, you can simplify the way you build the colnames below, with a single paste call with a 'sep="_"' argument, like:

colnames <- paste(embedding_key, 1:2, sep = "_")

Paste coerces numbers to strings.

Reply:
Thanks both! I changed it a bit following your suggestions, but keeping tSNE with low case "t" as this is what Seurat does by default.